File: backend/SindhDormitory.Domain/Entities/Room.cs
// =============================================================================
// Domain/Entities/Room.cs
// =============================================================================
using SindhDormitory.Domain.Enums;

namespace SindhDormitory.Domain.Entities;

public class Room : SoftDeletableEntity
{
    public int      RoomId        { get; set; }
    public int      FloorId       { get; set; }
    public string   RoomNumber    { get; set; } = string.Empty;
    public RoomType RoomType      { get; set; }
    public int      MaxOccupancy  { get; set; }
    public string?  Description   { get; set; }
    public bool     IsUnderMaintenance { get; set; } = false;

    // Navigation
    public Floor          Floor { get; set; } = null!;
    public ICollection<Bed> Beds  { get; set; } = [];
}

File: backend-node/src/entities/Room.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany, JoinColumn, CreateDateColumn, UpdateDateColumn } from 'typeorm';
import { Floor } from './Floor';
import { Bed } from './Bed';

@Entity('Rooms')
export class Room {
  @PrimaryGeneratedColumn({ name: 'RoomId' })
  roomId!: number;

  @Column({ name: 'FloorId', type: 'int' })
  floorId!: number;

  @Column({ name: 'RoomNumber', type: 'varchar', length: 20 })
  roomNumber!: string;

  @Column({ name: 'RoomType', type: 'varchar', length: 10, default: 'Double' })
  roomType!: string;

  @Column({ name: 'IsActive', type: 'boolean', default: true })
  isActive!: boolean;

  @Column({ name: 'IsDeleted', type: 'boolean', default: false })
  isDeleted!: boolean;

  @Column({ name: 'DeletedAt', type: 'datetime', nullable: true })
  deletedAt!: Date | null;

  @CreateDateColumn({ name: 'CreatedAt' })
  createdAt!: Date;

  @UpdateDateColumn({ name: 'UpdatedAt' })
  updatedAt!: Date;

  @ManyToOne(() => Floor, (f) => f.rooms, { onDelete: 'CASCADE' })
  @JoinColumn({ name: 'FloorId' })
  floor!: Floor;

  @OneToMany(() => Bed, (b) => b.room)
  beds!: Bed[];
}
